src/server.c

```diff
@@ -4867,25 +4894,58 @@ sds genRedisInfoStringLatencyStats(sds info, dict *commands) {
     return info;
 }
 
+static void addInfoSectionsToDict(dict *section_dict, char **sections, int len) {
+    for (int i = 0; i < len; i++){
+        sds section = sdsnew(sections[i]);
+        if (dictAdd(section_dict, section, NULL)==DICT_ERR)
+            sdsfree(section);
+    }
+}
+
+/* Create a dictionary according to the user input or call function 
+ * The output parameters will be set value in this function based on the argv parameter.
+ * The return value will be the dictionary for genRedisInfoString function */
+dict *genInfoSectionDict(robj **argv, int argc, int *out_all, int *out_everything, int *out_default) {
+    char *defSections[] = {
+        "server", "clients", "memory", "persistence", "stats", "replication",
+        "cpu", "module_list", "errorstats", "cluster", "keyspace", "latencystats"};
+
+    dict *section_dict = dictCreate(&stringSetDictType);
+    if (argc == 0) {
+        if (out_default) *out_default = 1;
+        addInfoSectionsToDict(section_dict, defSections, sizeof(defSections)/sizeof(*defSections));
```

Comment:
How about doing the default sections dict static and re-use it, instead of re-generating it every time?

Reply:
you mean only in the case of info with no args, right? 
Since when the user provided any inputs, we could modify the dict. 
Also, we'll need to make sure to skip this optimization from sentinel since it modifies that dict. 
And we'll need to add some trick to avoid releasing it when done. 

Do you think these complications are worth it?

Reply:
@oranagra Yes, I was only referring to the default list. It's probably a good idea this doesn't introduce a regression, because some clients use `INFO` quite extensively.

Reply:
@yossigo please have a look at my last commit.